include/kernel.h

```diff
@@ -5217,6 +5217,9 @@ static inline void k_cpu_idle(void)
  * Similar to k_cpu_idle(), but called with interrupts locked if operations
  * must be done atomically before making the CPU idle.
  *
+ * This has the effect invoking irq_unlock(key) atomically before invoking the
+ * power-saving operation.  Interrupts remain in that state on return.
```

Comment:
The first sentence is not true for Cortex-M: the interrupts are unlocked unconditionally before invoking the power-save function.

The second sentence is true for Cortex-M: the interrupt state is restored to the "key" passed in the function.

Reply:
Updated to basically incorporate the documentation from `arch_cpu_atomic_idle()`, changing it from prescriptive to descriptive and attempting to make more clear that this pairs with `irq_lock()` in exactly the same way that `irq_unlock()` does.

Is this better?